crates/runtime/benches/vector_search.rs

```diff
@@ -95,6 +97,19 @@ async fn vector_search_benchmarks() -> Result<(), String> {
     Ok(())
 }
 
+macro_rules! handle_error {
```

Comment:
Is there any specific reason why macro is used for handling error here, instead of a error handling function / method

Reply:
I found using macro is easier/more straight forward for this case where we have multiple places with similar logic that could be wrapped as a macro